examples/hello-triangle/main.rs

```diff
@@ -59,9 +61,18 @@ fn main() {
     });
 
     let vs_bytes = include_bytes!("shader.vert.spv");
-    let vs_module = device.create_shader_module(vs_bytes);
+    let mut vs_words = vec!();
+    for bytes4 in vs_bytes.chunks(4) {
+        vs_words.push(u32::from_le_bytes([bytes4[0], bytes4[1], bytes4[2], bytes4[3]]));
+    }
+    let vs_module = device.create_shader_module(&vs_words);
+
     let fs_bytes = include_bytes!("shader.frag.spv");
-    let fs_module = device.create_shader_module(fs_bytes);
+    let mut fs_words = vec!();
+    for bytes4 in fs_bytes.chunks(4) {
+        fs_words.push(u32::from_le_bytes([bytes4[0], bytes4[1], bytes4[2], bytes4[3]]));
```

Comment:
I don't think will this work on a big endian machine? Also, if willing to use unsafe transmute, can save a copy and allocation here by simply doing something like:
```
assert!(fs_bytes.len() % 4 == 0);
let fs_module = device.create_shader_module(transmute(fs.words));
```
you will also avoid endianness issues I believe

Reply:
this wouldn't solve the alignment problem